### ISSUE TITLE
Unable to upgrade dokku on Ubuntu 14.04

### ISSUE THREAD

**Issue by mekwall** [dokku/dokku#1880]
Trying to do `apt-get upgrade` but it is failing on the dpkg postinst script for dokku.

apt log:
```
Setting up dokku (0.4.13) ...
Purging old database entries in /usr/share/man...
Processing manual pages under /usr/share/man...
Purging old database entries in /usr/share/man/nl...
Processing manual pages under /usr/share/man/nl...
Purging old database entries in /usr/share/man/fr...
Processing manual pages under /usr/share/man/fr...
Purging old database entries in /usr/share/man/da...
Processing manual pages under /usr/share/man/da...
Purging old database entries in /usr/share/man/ru...
Processing manual pages under /usr/share/man/ru...
Purging old database entries in /usr/share/man/zh_TW...
Processing manual pages under /usr/share/man/zh_TW...
Purging old database entries in /usr/share/man/pt_BR...
Processing manual pages under /usr/share/man/pt_BR...
Purging old database entries in /usr/share/man/sv...
Processing manual pages under /usr/share/man/sv...
Purging old database entries in /usr/share/man/uk...
Processing manual pages under /usr/share/man/uk...
Purging old database entries in /usr/share/man/ja...
Processing manual pages under /usr/share/man/ja...
Purging old database entries in /usr/share/man/el...
Processing manual pages under /usr/share/man/el...
Purging old database entries in /usr/share/man/pt...
Processing manual pages under /usr/share/man/pt...
Purging old database entries in /usr/share/man/fi...
Processing manual pages under /usr/share/man/fi...
Purging old database entries in /usr/share/man/sl...
Processing manual pages under /usr/share/man/sl...
Purging old database entries in /usr/share/man/zh_CN...
Processing manual pages under /usr/share/man/zh_CN...
Purging old database entries in /usr/share/man/id...
Processing manual pages under /usr/share/man/id...
Purging old database entries in /usr/share/man/vi...
Processing manual pages under /usr/share/man/vi...
Purging old database entries in /usr/share/man/ko...
Processing manual pages under /usr/share/man/ko...
Purging old database entries in /usr/share/man/es...
Processing manual pages under /usr/share/man/es...
Purging old database entries in /usr/share/man/de...
Processing manual pages under /usr/share/man/de...
Purging old database entries in /usr/share/man/cs...
Processing manual pages under /usr/share/man/cs...
Purging old database entries in /usr/share/man/sr...
Processing manual pages under /usr/share/man/sr...
Purging old database entries in /usr/share/man/tr...
Processing manual pages under /usr/share/man/tr...
Purging old database entries in /usr/share/man/gl...
Processing manual pages under /usr/share/man/gl...
Purging old database entries in /usr/share/man/hu...
Processing manual pages under /usr/share/man/hu...
Purging old database entries in /usr/share/man/pl...
Processing manual pages under /usr/share/man/pl...
Purging old database entries in /usr/share/man/eo...
Processing manual pages under /usr/share/man/eo...
Purging old database entries in /usr/share/man/it...
Processing manual pages under /usr/share/man/it...
Purging old database entries in /usr/share/man/hr...
Processing manual pages under /usr/share/man/hr...
0 man subdirectories contained newer manual pages.
0 manual pages were added.
0 stray cats were added.
0 old database entries were purged.
User 'dokku' already exists
docker:x:999:git,mekwall,dokku
Setting up plugin directories
Migrating old plugins
Enabling all core plugins
Install all core plugins
Adding user dokku to group adm
dpkg: error processing package dokku (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 dokku
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

dpkg log:
```
2016-01-25 08:36:06 configure dokku:amd64 0.4.13 <none>
2016-01-25 08:36:06 status half-configured dokku:amd64 0.4.13
```

Removing/renaming `/var/lib/dpkg/info/dokku.postinst` makes it install without error and it seems to work as expected, but I reckon this is not the way to solve this. Also, when looking in `dokku.postinst` I found that there's only one place where it will do an `exit 1`, and that's when it's called with an unknown argument.

Running `dokku.postinst` manually works without it returning error status 1:
```
➜  ~  sudo bash /var/lib/dpkg/info/dokku.postinst configure
Purging old database entries in /usr/share/man...
Processing manual pages under /usr/share/man...
Purging old database entries in /usr/share/man/nl...
Processing manual pages under /usr/share/man/nl...
Purging old database entries in /usr/share/man/fr...
Processing manual pages under /usr/share/man/fr...
Purging old database entries in /usr/share/man/da...
Processing manual pages under /usr/share/man/da...
Purging old database entries in /usr/share/man/ru...
Processing manual pages under /usr/share/man/ru...
Purging old database entries in /usr/share/man/zh_TW...
Processing manual pages under /usr/share/man/zh_TW...
Purging old database entries in /usr/share/man/pt_BR...
Processing manual pages under /usr/share/man/pt_BR...
Purging old database entries in /usr/share/man/sv...
Processing manual pages under /usr/share/man/sv...
Purging old database entries in /usr/share/man/uk...
Processing manual pages under /usr/share/man/uk...
Purging old database entries in /usr/share/man/ja...
Processing manual pages under /usr/share/man/ja...
Purging old database entries in /usr/share/man/el...
Processing manual pages under /usr/share/man/el...
Purging old database entries in /usr/share/man/pt...
Processing manual pages under /usr/share/man/pt...
Purging old database entries in /usr/share/man/fi...
Processing manual pages under /usr/share/man/fi...
Purging old database entries in /usr/share/man/sl...
Processing manual pages under /usr/share/man/sl...
Purging old database entries in /usr/share/man/zh_CN...
Processing manual pages under /usr/share/man/zh_CN...
Purging old database entries in /usr/share/man/id...
Processing manual pages under /usr/share/man/id...
Purging old database entries in /usr/share/man/vi...
Processing manual pages under /usr/share/man/vi...
Purging old database entries in /usr/share/man/ko...
Processing manual pages under /usr/share/man/ko...
Purging old database entries in /usr/share/man/es...
Processing manual pages under /usr/share/man/es...
Purging old database entries in /usr/share/man/de...
Processing manual pages under /usr/share/man/de...
Purging old database entries in /usr/share/man/cs...
Processing manual pages under /usr/share/man/cs...
Purging old database entries in /usr/share/man/sr...
Processing manual pages under /usr/share/man/sr...
Purging old database entries in /usr/share/man/tr...
Processing manual pages under /usr/share/man/tr...
Purging old database entries in /usr/share/man/gl...
Processing manual pages under /usr/share/man/gl...
Purging old database entries in /usr/share/man/hu...
Processing manual pages under /usr/share/man/hu...
Purging old database entries in /usr/share/man/pl...
Processing manual pages under /usr/share/man/pl...
Purging old database entries in /usr/share/man/eo...
Processing manual pages under /usr/share/man/eo...
Purging old database entries in /usr/share/man/it...
Processing manual pages under /usr/share/man/it...
Purging old database entries in /usr/share/man/hr...
Processing manual pages under /usr/share/man/hr...
0 man subdirectories contained newer manual pages.
0 manual pages were added.
0 stray cats were added.
0 old database entries were purged.
User 'dokku' already exists
docker:x:999:git,mekwall,dokku
Setting up plugin directories
Migrating old plugins
Enabling all core plugins
Install all core plugins
Adding user dokku to group adm
```

The machine is a VM hosted by Digital Ocean.

**Comment by josegonzalez** [comment]
It's possible a subcommand failed. What version are you upgrading from and what version is now installed?

**Comment by mekwall** [comment]
From 0.4.5 to 0.4.13. Anything else you need?

**Comment by sweiler** [comment]
Same issue here. I'm updating from 0.4.10 to 0.4.13, and my log shows:
```
Unpacking dokku (0.4.13) over (0.4.10) ...
reload: Job is not running: nginx
dpkg: warning: subprocess old post-removal script returned error exit status 1
dpkg: trying script from the new package instead ...
reload: Job is not running: nginx
dpkg: error processing archive /var/cache/apt/archives/dokku_0.4.13_amd64.deb (--unpack):
subprocess new post-removal script returned error exit status 1
```

**Comment by mekwall** [comment]
@sweiler Which version of nginx are you running?

**Comment by sweiler** [comment]
nginx is running at version 1.8.1.

**Comment by josegonzalez** [comment]
@sweiler your issue is different (the logs don't even look the same).

**Comment by josegonzalez** [comment]
@mekwall Dont know if you still have this broken server, but if so, can you do the following:

```
# clear out your old dokku package cache
dpkg -r dokku

# reinstall
TRACE=1 apt-get install -f dokku
```

And gist the output?

**Comment by royalaid** [comment]
I am having a similar problem, running `dpkg -r dokku` just prints
```
dpkg: error processing package dokku (--remove):
  subprocess installed pst-removal script returned error exit status 1
Errors were encountered while processing:
  dokku
```

**Comment by josegonzalez** [comment]
@royalaid can you do as I asked @mekwall in the comment right above yours?

**Comment by royalaid** [comment]
```
hook ~: dpkg -r dokku
(Reading database ... 97917 files and directories currently installed.)
Removing dokku (0.4.13) ...
reload: Unknown instance:
dpkg: error processing package dokku (--remove):
 subprocess installed post-removal script returned error exit status 1
Errors were encountered while processing:
 dokku
hook ~: TRACE=1 apt-get install dokku
Reading package lists... Done
Building dependency tree
Reading state information... Done
dokku is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
Need to get 0 B/38.1 kB of archives.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
dpkg: error processing package dokku (--configure):
 package dokku is not ready for configuration
 cannot configure (current status `half-installed')
Errors were encountered while processing:
 dokku
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by josegonzalez** [comment]
TRACE=1 apt-get install --reinstall dokku

**Comment by royalaid** [comment]
https://gist.github.com/royalaid/c4c3d64b4677785a5577

**Comment by josegonzalez** [comment]
[This should fix it](https://github.com/dokku/dokku/commit/174b5b74ad33830269adf5a4cb38817ddb06b804), though it's not in a release just yet. I'll do that tonight.

**Comment by royalaid** [comment]
kk, many thanks

**Comment by mekwall** [comment]
@josegonzalez Sorry for the late response! I recently reinstalled my vm and it is currently working. Maybe it had something to do with that I was using nginx own APT repo which uses the `nginx` package instead of `nginx-common`.

**Comment by josegonzalez** [comment]
The nginx bug should be fixed in 0.4.14.
